src/helpers/App.php

```diff
@@ -65,11 +65,35 @@ class App
      * @return string|array|false The environment variable value
      * @since 3.4.18
      */
-    public static function env(string $name)
+    public static function env(string $name): array|string|false
     {
         return $_SERVER[$name] ?? getenv($name);
     }
 
+    /**
+     * Returns the value of a constant, falling back to an environment variable.
+     *
+     * @param string $constName The name of the constant
+     * @param string|null $envName The name of the environment variable, if different from $constName
+     * @param bool $bool Whether to parse the value as a boolean
+     * @return mixed The value of the constant or environment variable, or null if neither are found
+     * @since 4.0.0
+     */
+    public static function constant(string $constName, ?string $envName = null, bool $bool = false): mixed
+    {
+        if (defined($constName)) {
+            return constant($constName);
+        }
+
+        $envVal = static::env($envName ?? $constName);
+
+        if ($envVal === false) {
+            return null;
+        }
+
+        return $bool ? filter_var($envVal, FILTER_VALIDATE_BOOL, FILTER_NULL_ON_FAILURE) : $envVal;
+    }
+
```

Comment:
I considered calling this `App::config` or something, and rolling in the `argv` checks we're doing in bootstrap…